src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/LeonardoService.scala

```diff
@@ -131,6 +131,25 @@ class LeonardoService(protected val dataprocConfig: DataprocConfig,
     }
   }
 
+  // Meant to be run synchronously to the cluster creation API call
+  // i.e. We wait for the method to finish before sending the API response
```

Comment:
Pedantic, but technically it still returns a Future so it's still async in some sense. Maybe the comment could read "We complete the API response with the result of the returned Future"